src/utils/dataset_utils.py

```diff
@@ -117,113 +120,172 @@ def match_val_targets_location(val,targets):
+                # columns[3] is a string of comman separated alternate spellings
+                for key in columns[3].lower().split(','):
+                    cities[key].add(value)
+                    
+                    # Locations from urls are often run together
+                    cities[key.replace(' ', '')].add(value)
+
+        self.cities = cities
 
-def clean_url(text):
+    def fast_loc(self, span_input):
+        """
+        Uses geotext and data from the geoname website to quickly detect if input is a locaiton
+    
+        string text: text to test for location
+        """       
+
+        span = span_input.get_span()
+        reg = re.compile(r'[^a-zA-Z ]')
+
+        if len(span) < 4 or reg.search(span):
+            return False
+
+        city = True if self.cities[span.lower()] else False
+        state = span.title() in lookup_state_name(span)
+        country = span.title() in lookup_country_name(span)
+
+        return city or state or country
+
+def fix_spacing(text):
+    """
+    Removes double spaces and spaces at the beginning and end of line
+    
+    string text: text to fix spacing on
+    """    
+    
+    while '  ' in text:
+        text = text.replace('  ',' ')
+    text = text.strip()
+    
+    return text
+    
+def parse_url(text):
     """
     Finds location mentions in url and labels them.
     
     string text: string of url
     """
-        
-    text = text.replace('https', '').replace('http', '').replace(':', ' ').replace('/', ' ').replace('.', ' ')
-    text = text.replace('backpage', '').replace('com', '').replace('online', '')
-    text = text.replace('  ', ' ').replace('  ', ' ')
-    text = text.title().split()
-                
-    url_locs = ""
-    for word in text:
-        places = geograpy.get_place_context(text=word)
-        if places.cities:
-            url_locs += " The city of " + str(places.cities[0]) + " (url)."
-        if places.regions:
-            url_locs += " The region of " + str(places.regions[0]) + " (url)."
-        if places.other:
-            url_locs += " The location of " + str(places.other[0]) + " (url)."
-     
-    return url_locs
-
-def clean_html(text):
+    
+    text = clean_input(text)
+    
+    # Create spacing
+    text = text.replace('https', '').replace('http', '').replace('com', '')
+    text = text.replace(':', ' ').replace('/', ' ').replace('\\', ' ').replace('.', ' ').replace('-', ' ')
+    
+    # URL_S marks start of word in URL, URL_E marks the end
+    url = 'Url ' + text.title() + '. '
+    
+    url = fix_spacing(url)
+    
+    return url
+
+def clean_input(text):
     """
-    Removes html specific tags from memex_raw_data
+    Removes quotes, html tags, etc
     
     string text: string to modify
     """
     
-    # Work in progress
-    text = text.replace('\n',' ').replace('\r',' ').replace('  ',' ').replace('  ',' ')
-    reg = re.compile(r'<div class="postingBody">.*?<div class=')
-    match = reg.search(text)
```

Comment:
Is this all replaced somewhere now (`clean_html` and `clean_url`)?

Reply:
Yes, clean_input handles that.